keps/sig-apps/1591-daemonset-surge/README.md

```diff
@@ -141,6 +145,30 @@ pod). If the old pod delays deletion, then the new pod has a chance to accept ha
   * Testing should set up conflicting rules like HostPort and verify that surge fails and the correct daemonset condition is set and events are generated.
 	* A test should cover a pod going unready during rollout and verifying it is immediately replaced.
 
+
+#### Prerequisite testing updates
+[x] I/we understand the owners of the involved components may require updates to existing tests to make this code solid enough prior to committing the changes necessary to implement this enhancement.
+
+##### Unit tests
+
+```
+`k8s.io/kubernetes/pkg/apis/apps/validation`                     `06/06/2022`:     `90.6% of statements` `The tests added for the current feature in this package touches the daemonSet Spec field. No new tests are needed for promotion to GA`
+`k8s.io/kubernetes/pkg/apis/apps/validation/validation.go:387`:	 `06/06/2022`:		`100.0% of statements`
+`k8s.io/kubernetes/pkg/controller/daemon`: `06/06/2022`: `70.7% of statements`    `The tests added for the current feature in this package touches the daemonSet update strategies. No new tests are needed for promotion to GA`
+`k8s.io/kubernetes/pkg/registry/apps/daemonset`: `06/06/2022`: `31.1% of statements`  `The tests added for the current feature in this package makes sure that the kubernetes version upgrades/downgrades won't have any impact on the new field to the daemonSet api when persisting to etcd. No new tests are needed for promotion to GA`
+`k8s.io/kubernetes/pkg/registry/apps/daemonset/strategy.go:129`:	`06/06/2022`:	`100.0% of statements`
+```
+##### Integration tests
+
+Will add a new integration test to test the daemonSet upgrade/downgrade strategies.
+
+##### e2e tests
+
+An e2e test which exercises maxSurge when surge is used as update strategy is added for daemonsets.
+
+- should surge pods onto nodes when spec was updated and update strategy is RollingUpdate: [test grid](https://testgrid.k8s.io/sig-apps#gce-serial)
```

Comment:
Please link to the traige page:

https://storage.googleapis.com/k8s-triage/index.html


so that we can easily assess its flakiness